components/canvas/webgl_thread.rs

```diff
@@ -273,8 +314,13 @@ impl<VR: WebVRRenderHandler + 'static> WebGLThread<VR> {
         // It is important that the fence sync is properly flushed into the GPU's command queue.
         // Without proper flushing, the sync object may never be signaled.
         data.ctx.gl().flush();
-
-        (info.texture_id, info.size, gl_sync as usize)
+        // If no Swap message received we use the currently bound IOSurface
+        if !info.has_request_animation {
+            info.io_surface_id = data.ctx.get_active_io_surface_id();
+        } else {
+            info.io_surface_id = data.ctx.handle_lock();
```

Comment:
This is one place where it is not clear to me if my plan to only store a single TextureBacking value in WebGLContextInfo will be enough.